app/questionnaire/value_source_resolver.py

```diff
@@ -116,6 +121,34 @@ def _resolve_answer_value_source(
 
         return answer_value
 
+    def _resolve_progress_value_source(
+        self, value_source: Mapping
+    ) -> ValueSourceEscapedTypes | ValueSourceTypes | None:
+        identifier = value_source["identifier"]
+        selector = value_source["selector"]
+        if selector == "section":
+            return self.progress_store.get_section_status(
+                section_id=identifier, list_item_id=None
+            )
+
+        if selector == "block":
+            if not self.location:
+                raise ValueError("location is required to resolve block progress")
+
+            if self.location.section_id == self.schema.get_section_id_for_block_id(
+                identifier
+            ) and not self._is_block_on_path(identifier):
+                return None
+
+            if section_id := self.schema.get_section_id_for_block_id(identifier):
+                return self.progress_store.get_block_status(
+                    block_id=identifier,
+                    section_id=section_id,
+                    list_item_id=self.location.list_item_id
+                    if self.location.section_id == section_id
+                    else None,
+                )
```

Comment:
Assign section id to a var since it might be called twice.

```py
            section_id_for_block = self.schema.get_section_id_for_block_id(identifier)
            if (
                self.location.section_id == section_id_for_block
                and not self._is_block_on_path(identifier)
            ):
                return None

            return self.progress_store.get_block_status(
                block_id=identifier,
                section_id=section_id_for_block,
                list_item_id=self.location.list_item_id
                if self.location.section_id == section_id_for_block
                else None,
            )
```

- The `if section_id` is not required?. Was it just for type hints?.
- Why do we have `self.location.section_id == self.schema.get_section_id_for_block_id(
                identifier
            )` does this not mean if the block was in a different section, we are not checking it is on the path?. We must not have a test for it?

Reply:
Updated this method as above, need to check on the second point